src/ranged.cpp

```diff
@@ -1221,7 +1221,7 @@ std::vector<tripoint> game::pl_target_ui( target_mode mode, item *relevant, int
                            m.mode.c_str(), m.qty );
             }
 
-            auto cur = ammo ? ammo : m->ammo_data();
+            const itype *cur = ammo ? ammo : m->ammo_data();
```

Comment:
> variable named str that is the result of string_format is a string?

Well, yes and it's probably acceptable here but the general convention in this repo is to omit using ```auto``` as much as possible and keep the for iterators.